### ISSUE TITLE
SCM: use better aria label for status bar item

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#114361]
1. Turn on a screen reader
2. Press F6 until focus is on the status bar
3. Use left / right arrow to pass focus to the SCM Checkout Branch
4. Screen reader reads: "Checkout branch/tag"

Or simply dev tools investigate scm status bar item and notice the `aria-label` proprety.

The aria label should read the name of the branch.
I know this was working before thus adding regression.

Please note that we have nice accessibility API so you should be able to nicely contribute an aria-label for your status bar item. The aria label should be nice and concise and contain the branch name

Found by @jvesouza

**Comment by eamodio** [comment]
@isidorn Why is this marked as a regression? Nothing has changed here.

What is our pattern here? This status bar entry is doing double duty in that it is showing a bunch of status in its icon, text, and tooltip. How much should is say? Should it be something like `<repo name> (<scm-provider>), branch <branch-name>. Click to <command-title>`, e.g. `vscode (Git), branch master. Click to Checkout branch/tag...`

Currently we are somewhat limited here in what the SCM provider api provides -- it doesn't actually expose status bar items, instead it provides a `Command` object that gets turned into a status bar item. Another complication is that the "text" of the status bar item (which is the `command.title`) could (and does in Git) contain codicons -- do we need to strip those?

**Comment by isidorn** [comment]
@eamodio I thought this was working before - at least that is what our users thought as well. Feel free to remove the regression label if I was mistaken.

The aria label should be concise. I suggest for your example just
`vscode, master`
Alternative is
`vscode, master branch`
Always put the least important information at the end so it can be skipped by users.

Yes codicons should be stripped from aria labels if possible.

@jvesouza might have ideas on best aria label to user here

**Comment by eamodio** [comment]
So without API changes, we won't be able to shorten it to something like you suggest. So maybe we can get close as a fallback, but also add a new property on the SCM contributed commands similar to what we added to tree items.

**Comment by jvesouza** [comment]
For me the problem is that I can't identify through the VSCode which branch I am working on.
I end up having to run the git status command externally to identify the branch.
It seems to me that the most reasonable place to find this information would be the status bar, but if there is another way to obtain this information it will also work.
Thanks.

**Comment by isidorn** [comment]
@eamodio okey, if there is a way to include the name of the branch without API changes that would also work.
@jvesouza yeah I think it makes sense this information is in the status bar, as for alternative places where we might have this @eamodio would be better to answer

**Comment by eamodio** [comment]
I did the best I could without API changes here, so with the Git extension you will see something like the following aria labels:

`master, vscode (Git) - Checkout branch/tag...` for the "branch" status bar entry
'vscode (Git) - Synchronize Changes' for the publish/sync status bar entry (though the "Synchronize Changes" could be different based on if its a publish vs a sync, etc)

**Comment by isidorn** [comment]
@eamodio thanks, this sounds good to me!

**Comment by jvesouza** [comment]
This sounds good to me.
Thanks.

**Comment by isidorn** [comment]
Adding verified per @jvesouze comment
